pkg/ddl/executor.go

```diff
@@ -4554,6 +4512,9 @@ func (e *executor) CreatePrimaryKey(ctx sessionctx.Context, ti ast.Ident, indexN
 				return dbterror.ErrUniqueKeyNeedAllFieldsInPf.GenWithStackByArgs("PRIMARY")
 			}
 			// index columns does not contain all partition columns, must set global
+			if indexOption == nil || !indexOption.Global {
+				return dbterror.ErrGlobalIndexNotExplicitlySet.GenWithStackByArgs("PRIMARY")
+			}
 			global = true
```

Comment:
Param global can be removed, just like createIndex

Reply:
Nice find, I removed it and also removed the handling of the 'global' arg in the `job.Args`/`decodeAddIndexArgs`